test/server.test.js

```diff
@@ -4,14 +4,179 @@ import request from 'supertest';
+      request(app)
+        .get('/superlative/name')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    }); 
+
+    // Add test for POST to /addSuperlative without modifying the db
+    
+    it('should respond with status 200 when GET is sent to /getAllSuperlatives', (done) => {
+      request(app)
+        .get('/getAllSuperlatives')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    }); 
+
+    it('should respond with status 200 when POST is sent to /getParticularSuperlative', (done) => {
+      request(app)
+        .post('/getParticularSuperlative')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    });
+  });
+
+  describe('Student path', () => {
+    it('should respond with status 200 when GET is sent to /students', (done) => {
+      request(app)
+        .get('/students')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    }); 
+
+    // Add test for POST to /addStudent without modifying the db
+
+    it('should respond with status 200 when GET is sent to /getAllStudents', (done) => {
+      request(app)
+        .get('/getAllStudents')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    });
+
+    it('should respond with status 404 when body-less GET is sent to /getStudent/id', (done) => {
+      request(app)
+        .get('/getStudent/id')
+        .expect(404)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    });
+
+    // Add test for successful POST to /getParticularStudent without modifying the db
+
+    it('should respond with status 404 when body-less POST is sent to /getParticularStudent', (done) => {
+      request(app)
+        .get('/getParticularStudent')
+        .expect(404)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    }); 
+  });
+
+  describe('Superlative requests path', () => {
+    // Add test for successful POST to /addShoutout without modifying the db
+  
+    it('should respond with status 200 when GET is sent to /getAllShoutouts', (done) => {
+      request(app)
+        .get('/getAllShoutouts')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    });
+
+    it('should respond with status 200 POST is sent to /getParticularShoutout', (done) => {
+      request(app)
+        .post('/getParticularShoutout')
+        .expect(200)
+        .end((err, res) => {
+          if (err) return done(err);
+          done();
+        });
+    }); 
+
+    it('should respond with status 200 when POST is sent to /getParticularShoutout', (done) => {
```

Comment:
Do we need both this test and the one just before it? (which doesn't include the req body data)

Reply:
Probably not, I changed it because it wasn't sending a 404 but I'll update the PR to remove it